models/articles.models.js

```diff
@@ -11,13 +11,25 @@ exports.fetchArticles = (sort_by, order) => {
         return Promise.reject({ status: 400, msg: "bad request" });
     }
     
-    let query = format(`SELECT articles.author, articles.title, articles.article_id, articles.topic,articles.created_at, articles.votes, articles.article_img_url, 
-    COUNT(comments.comment_id) AS comment_count FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id ORDER BY %I %s`, sort_by, order.toUpperCase()
-    );
+    let query = format(`
+        SELECT articles.author, articles.title, articles.article_id, articles.topic,articles.created_at, articles.votes, articles.article_img_url, 
+        COUNT(comments.comment_id) AS comment_count 
+        FROM articles
+        LEFT JOIN comments ON articles.article_id = comments.article_id`);
+
+    if (topic) {
+        query += format(` WHERE articles.topic = %L`, topic);  
+    }
+
+    query += format(`
+        GROUP BY articles.article_id
+        ORDER BY %I %s;`, sort_by, order.toUpperCase());
 
     return db.query(query).then(({rows}) => {
+        
+        if(rows.length === 0){
+            return Promise.reject({ status: 204, msg: 'No content' })
```

Comment:
I would specify here that the topic hasn't been found. I would add different functionality to make sure that if a topic exists but doesn't have a any articles I return a 200 with an empty array